lib/blob.js

```diff
@@ -359,19 +604,496 @@ Blob.prototype.getContainerMetadata = function getContainerMetadata(name) {
+ *                                      // or blob (public read access for blobs)
+ *                                      // If it is not specified, the resource will be private and will be accessed only by the account owner
+ *    accessPolicies: [{                // The container ACL settings. An array with five maximum access policies objects (optional)
+ *      id:     '...',                  // Unique identifier, up to 64 chars in length
+ *      start:  new Date(),             // Time from which access policy is valid
+ *      expiry: new Date(),             // Expiration of access policy
+ *      permission: {                   // Set of permissions delegated
+ *        read:              false,     // Read the content, properties, metadata or block list of a blob or of
+ *                                      // any blob in the container if the resourceType is a container.
+ *        add:               false,     // Add a block to an append blob or to any append blob if the resourceType is a container.
+ *        create:            false,     // Write a new blob, snapshot a blob, or copy a blob to a new blob. These operations can be done to any blob in the container
+ *                                      // if the resourceType is a container.
+ *        write:             false,     // Create or write content, properties, metadata, or block list. Snapshot or lease the blob. Resize the blob (page blob only).
+ *                                      // These operations can be done for every blob in the container if the resourceType is a container
+ *        delete:            false,     // Delete the blob or any blob in the container if the resourceType is a container.
+ *        list:              false,     // List blobs in the container.
+ *      }
+ *    }]
+ * }
+ * ```
+ * @param {string} leaseId - GUID string; lease unique identifier (optional)
+ * @returns {Promise} A promise that the permissions have been set
+ */
+Blob.prototype.setContainerACL = function setContainerACL(name, options, leaseId) {
+  assert(name, 'The name of the container must be specified');
+  var query = {
+    restype: 'container',
+    comp: 'acl'
+  };
+  var path = '/' + name;
+  var headers = {};
+  if (leaseId) {
+    assert(utils.isValidGUID(leaseId), '`leaseId` is not a valid GUID.');
+    headers['x-ms-lease-id'] = leaseId;
+  }
+
+  if (options.publicAccessLevel){
+    assert(options.publicAccessLevel === 'container' || options.publicAccessLevel === 'blob',
+      "The supplied `publicAccessLevel` is incorrect. The possible values are: container and blob")
+      headers['x-ms-blob-public-access'] = options.publicAccessLevel;
+  }
+  if (options.accessPolicies && options.accessPolicies.length > 5){
+    throw new Error("The supplied access policy is wrong. The maximum number of the access policies is 5");
+  }
+
+  // Construct the payload
+  var data = '<?xml version="1.0" encoding="utf-8"?>';
```

Comment:
Generating XML using string manipulation makes me a little nervous in general.  Since you're validating the the policy id and everything else is is is generated in a bounded way, this is fine.

If this weren't so strictly bounded, I would want us to either generate the XML using an XML library, or at least parse and validate the resulting XML using one.